File: asm-lens/example/math_utils.c
#include "math_utils.h"

int square(int x) {
    return x * x;
}

int sum_squares(int n) {
    int total = 0;
    for (int i = 1; i <= n; i++) {
        total += square(i);
    }
    return total;
}

File: example/fortran/math_utils.f90
module math_utils
    implicit none
contains

    pure function square(x) result(res)
        integer, intent(in) :: x
        integer :: res
        res = x * x
    end function square

    pure function sum_squares(n) result(total)
        integer, intent(in) :: n
        integer :: total
        integer :: i
        total = 0
        do i = 1, n
            total = total + square(i)
        end do
    end function sum_squares

end module math_utils
